### PR TITLE
[FEATURE] a basic RWrapper class

### PR DESCRIPTION
**Pull request by cbielow**
with example usage for InternalCalibration.

For the example:
R code is currently placed in share/OpenMS/SCRIPTS, which has the advantage that any user can modify the plots to their need (as opposed to hard-coding the script in C++ strings).

The wrapper can certainly be extended, but the error handling up to this point is quite ok (i.e. the user 
is enabled to fix the problem).

### CODE REVIEW

**Comment by cbielow** [comment]
if everything goes as expected, InternalCalibration will print 
```
...
Running R script 'M:/dev/oms_pick/share/OpenMS/SCRIPTS/InternalCalibration_Models.R' ... success
Running R script 'M:/dev/oms_pick/share/OpenMS/SCRIPTS/InternalCalibration_Residuals.R' ... success
```

**Comment by cbielow** [comment]
and produce plots like 
models
residuals

(if requested via `-quality_control:models_plot` and `-quality_control:residuals_plot`)

File: src/openms/include/OpenMS/SYSTEM/File.h [hunk 1]
```diff
@@ -183,13 +182,53 @@ namespace OpenMS
...
+      This is useful if you have an optional
+      output file, which may, or may not be requested, but you need its content regardless,
+      e.g. for intermediate plotting with R.
+      Thus you can just call this function to get a file which can be used and gets automatically
+      destroyed if needed.
+
+      @param alternative_file If this string is not empty, no action is taken and it is used as return value
+      @return Full path to a temporary file
+    */
+    static const String& getTemporaryFile(const String& alternative_file = "");
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/SYSTEM/File.h]: some suggestion:
- make TemporaryFile a class that has a destructor that removes the file. That way the user can control if he wants to delete the file in a local context or on program exit.
- the behavior of returning the alternative file name is - in my view -  trying to be smart but is imho less transparent than just checking for empty string at the actual site of useage

File: src/openms/include/OpenMS/SYSTEM/RWrapper.h [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Marc Sturm, Chris Bielow $
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/SYSTEM/RWrapper.h]: Marc Sturm?

File: src/openms/include/OpenMS/SYSTEM/RWrapper.h [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  */
+  class OPENMS_DLLAPI RWrapper
+  {
+public:
+
+ 
+    /**
+      @brief Look for an R script in the share/OpenMS/SCRIPT folder
+      
+      The script filename can be an absolute filename (in which case nothing happens),
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/SYSTEM/RWrapper.h]: nothing happens -> returns the filename as is

File: src/topp/InternalCalibration.cpp [hunk 4]
```diff
@@ -200,7 +203,7 @@ class TOPPInternalCalibration :
     //registerIntOption_("RANSAC:pc_n", "<# points>", 20, "Percentage (1-99) of initial model points from available data.", false);
     //setMinInt_("RANSAC:pc_n", 1);
     //setMaxInt_("RANSAC:pc_n", 99);
-    registerDoubleOption_("RANSAC:thresh", "<threshold>", 10.0, "Threshold for accepting inliers (ppm^2 distance)", false);
+    registerDoubleOption_("RANSAC:thresh", "<threshold>", 10.0, "Threshold for accepting inliers (instrument precision (not accuracy!) as ppm^2 distance)", false);
```
Comment (timosachsenberg) [inline src/topp/InternalCalibration.cpp]: thresh -> threshold

**Comment by timosachsenberg** [comment]
nice addition. I am thinking about moving the MetaProSIP plotting to your format with pre-generated scripts

**Comment by cbielow** [comment]
ok, this should do it

**Comment by timosachsenberg** [comment]
+1

**Pull request merged**